### PR TITLE
E2e

### PR DESCRIPTION
**Pull request by tombenezra**

### CODE REVIEW

**Comment by tombenezra** [comment]
1. currently there is one test that uses the binaries of the cli to test 3 different sites (eventually skipped the publish to local registry process due to its overhead). 
2. as a first step I attempt to login to wix.com, but credentials are exposed on the driver, might be a problem (suggestions are welcomed, maybe storing keys in CI or something)

File: .travis.yml [hunk 1]
```diff
@@ -14,6 +14,8 @@ before_install:
 
 install:
   - yarn
+  - export DISPLAY=':99.0'
+  - Xvfb :99 -screen 0 1024x768x24 > /dev/null 2>&1 &
```
Comment (dor-itzhaki) [inline .travis.yml]: those configuration are already defined (see services->xvfb) under linux tests.
i don't think we need this commit

File: packages/corvid-e2e-tests/.eslintrc [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (dor-itzhaki) [inline packages/corvid-e2e-tests/.eslintrc]: we no longer have local eslint config files
you can just delete this file, assuming the global one fits

File: packages/corvid-e2e-tests/.eslintignore [hunk 3]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (dor-itzhaki) [inline packages/corvid-e2e-tests/.eslintignore]: we no longer have local eslintignore files

File: packages/corvid-e2e-tests/jest.setup.js [hunk 4]
```diff
@@ -0,0 +1 @@
+jest.setTimeout(150000);
```
Comment (dor-itzhaki) [inline packages/corvid-e2e-tests/jest.setup.js]: in corvid-cli, this file is called setupJest.js

File: packages/corvid-e2e-tests/package.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
+{
+  "name": "corvid-e2e-tests",
+  "version": "1.0.0",
```
Comment (dor-itzhaki) [inline packages/corvid-e2e-tests/package.json]: should be aligned to the current module version (in lerna.json)

File: packages/corvid-e2e-tests/package.json [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  "description": "",
+  "main": "jest.setup.js",
```
Comment (dor-itzhaki) [inline packages/corvid-e2e-tests/package.json]: do we have a main file ?

File: packages/corvid-e2e-tests/package.json [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  "scripts": {
+    "test": "jest --detectOpenHandles",
+    "precommit": "lint-staged"
```
Comment (dor-itzhaki) [inline packages/corvid-e2e-tests/package.json]: no local lint

File: packages/corvid-e2e-tests/package.json [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  },
+  "devDependencies": {
```
Comment (dor-itzhaki) [inline packages/corvid-e2e-tests/package.json]: no need for local eslint, husky, lint-staged, etc...

File: packages/corvid-e2e-tests/package.json [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+  "jest": {
+    "testEnvironment": "node",
+    "testMatch": [
+      "**/*.+(e2e).js"
+    ],
+    "setupFilesAfterEnv": [
+      "<rootDir>/jest.setup.js"
+    ]
+  },
+  "husky": {
```
Comment (dor-itzhaki) [inline packages/corvid-e2e-tests/package.json]: remove

File: packages/corvid-e2e-tests/package.json [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    "hooks": {
+      "precommit": "lint-staged"
+    }
+  },
+  "lint-staged": {
```
Comment (dor-itzhaki) [inline packages/corvid-e2e-tests/package.json]: remove

File: packages/corvid-e2e-tests/package.json [hunk 11]
```diff
@@ -0,0 +1,48 @@
+{
+  "name": "corvid-e2e-tests",
```
Comment (dor-itzhaki) [inline packages/corvid-e2e-tests/package.json]: corvid-cli-e2e-tests ?

File: packages/corvid-e2e-tests/package.json [hunk 12]
```diff
@@ -0,0 +1,48 @@
+{
```
Comment (dor-itzhaki) [inline packages/corvid-e2e-tests/package.json]: do we really want it to be a separate module ?

File: packages/corvid-cli/src/commands/open-editor.js [hunk 13]
```diff
@@ -122,10 +126,17 @@ module.exports = {
   command: "open-editor",
   describe: getMessage("OpenEditor_Command_Description"),
   builder: args =>
-    args.option("ignore-certificate", {
-      describe: "ignore certificate errors",
-      type: "boolean"
-    }),
+    args
+      .option("ignore-certificate", {
```
Comment (dor-itzhaki) [inline packages/corvid-cli/src/commands/open-editor.js]: we can add those args to all commands by default by using yargs.middleware

File: packages/corvid-e2e-tests/test/drivers/localEditorDriver.js [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+  const addTextElement = async () => {
+    await page.click(ADD_ELEMENT_SELECTOR);
+    await page.waitForSelector(ADD_TEXT_SELECTOR);
+    await page.hover(ADD_TEXT_SELECTOR);
+    await page.waitForSelector(THEMED_TEXTS_SELECTOR);
+    const textElements = await page.$(THEMED_TEXTS_SELECTOR);
+    await textElements.click();
+  };
+
+  const login = async () => {
```
Comment (dor-itzhaki) [inline packages/corvid-e2e-tests/test/drivers/localEditorDriver.js]: user & pass should probably be given as args.
the test can take them from the environment if you don't want them exposed

File: packages/corvid-e2e-tests/test/sanity.e2e.js [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+    await login(cliDriver);
+  });
+
+  testSites.forEach(({ editorUrl, description }) =>
+    test(`should clone ${description} site, open it and push without making actual changes`, async () => {
+      const cleanup = initTempDirectory();
+      const corvidCloneCmd = await cliDriver.clone({ editorUrl });
+      expect(corvidCloneCmd.code).toEqual(EXIT_CODE_SUCCESS);
+      const remoteDebuggingPort = await findAvailablePort();
+      cliDriver.openEditor({ remoteDebuggingPort });
```
Comment (dor-itzhaki) [inline packages/corvid-e2e-tests/test/sanity.e2e.js]: maybe openEditor can:
1. return the debug browserURL
2. return a promise that is resolved only once the editor is open

File: packages/corvid-e2e-tests/test/sanity.e2e.js [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+    return () => process.chdir(cwd);
+  };
+
+  beforeEach(async () => {
+    await cliDriver.logout();
+    await login(cliDriver);
+  });
+
+  testSites.forEach(({ editorUrl, description }) =>
+    test(`should clone ${description} site, open it and push without making actual changes`, async () => {
```
Comment (dor-itzhaki) [inline packages/corvid-e2e-tests/test/sanity.e2e.js]: right now this only checks that cloning and pushing pass without errors, but we don't actually test their "output" (files on the hard-drive)

**Pull request merged**